pkg/networkservice/chains/nsmgr/server_test.go

```diff
@@ -169,6 +169,64 @@ func TestNSMGR_SelectsRestartingEndpoint(t *testing.T) {
 	require.NoError(t, ctx.Err())
 }
 
+func TestNSMGR_SettingPayload(t *testing.T) {
```

Comment:
Probably it would be better to check this via registry api. For this, you need to setup domain with only registry and check that `setpayload` is working correctly via Find API. Also, this test can be moved to `registry/chains/memory`